packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js

```diff
@@ -134,36 +127,42 @@ class ActionDropdown extends React.Component {
 	constructor(props) {
 		super(props);
 		this.onToggle = this.onToggle.bind(this);
-		this.state = {
-			dropup: props.dropup,
-		};
+		this.state = {};
 	}
 
-	onToggle(isOpen, event) {
-		if (!isOpen) {
-			this.setState({ dropup: this.props.dropup });
-			return;
-		}
-
-		const dropdownTrigger = getDropdownToggleFromInner(event.target);
-		const dropdownMenu = dropdownTrigger.nextSibling;
-		const dropdownContainer = getDropdownContainer(dropdownTrigger);
+	componentDidUpdate(prevProps, prevState) {
+		/*
+		Dropdown/Dropup automatic switch:
+		depending on its position with a defined container, it will switch from down and up mode.
 
-		if (dropdownContainer) {
-			const dropdownRect = dropdownMenu.getBoundingClientRect();
-			const containerRect = dropdownContainer.getBoundingClientRect();
+		By default it checks its position with the <body>.
+		Specific container support with "tc-dropdown-container" classname on the parent container.
+		 */
+		if (!prevState.isOpen && this.state.isOpen) {
+			// eslint-disable-next-line react/no-find-dom-node
+			const dropdown = ReactDOM.findDOMNode(this.ref);
+			const dropdownTrigger = dropdown.querySelector('.dropdown-toggle');
+			const dropdownMenu = dropdownTrigger.nextSibling;
+			const dropdownContainer = getDropdownContainer(dropdownTrigger);
 
-			this.setState(oldState => {
-				if (!oldState.dropup && dropdownRect.bottom > containerRect.bottom) {
-					return { dropup: true };
-				} else if (oldState.dropup && dropdownRect.top < containerRect.top) {
-					return { dropup: false };
+			if (dropdownContainer) {
+				console.log('container found', dropdownContainer.tagName);
```

Comment:
Did you leave those on purpose ? Plus, it adds error to the linting report (above)

Reply:
Oh no, thanks

Reply:
+ I'll suppress the other linter errors as I haven't found any other way to do that